diff --git a/Cargo.toml b/Cargo.toml
index 71398530f..6560e810d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -68,6 +68,7 @@ members = [
     "testing/ef_tests",
     "testing/execution_engine_integration",
     "testing/node_test_rig",
+    "testing/proof_engine",
     "testing/simulator",
     "testing/state_transition_vectors",
     "testing/validator_test_rig",
@@ -141,7 +142,7 @@ eth2_network_config = { path = "common/eth2_network_config" }
 eth2_wallet = { path = "crypto/eth2_wallet" }
 ethereum_hashing = "0.8.0"
 ethereum_serde_utils = "0.8.0"
-ethereum_ssz = { version = "0.10.0", features = ["context_deserialize"] }
+ethereum_ssz = { version = "0.10.4", features = ["context_deserialize"] }
 ethereum_ssz_derive = "0.10.0"
 execution_layer = { path = "beacon_node/execution_layer" }
 filesystem = { path = "common/filesystem" }
@@ -219,7 +220,7 @@ slashing_protection = { path = "validator_client/slashing_protection" }
 slot_clock = { path = "common/slot_clock" }
 smallvec = "1"
 snap = "1"
-ssz_types = { version = "0.14.0", features = ["context_deserialize", "runtime_types"] }
+ssz_types = { version = "0.14.1", features = ["context_deserialize", "runtime_types"] }
 state_processing = { path = "consensus/state_processing" }
 store = { path = "beacon_node/store" }
 strum = { version = "0.27", features = ["derive"] }
@@ -275,4 +276,3 @@ debug = true
 
 [patch.crates-io]
 quick-protobuf = { git = "https://github.com/sigp/quick-protobuf.git", rev = "87c4ccb9bb2af494de375f5f6c62850badd26304" }
-
