diff --git a/beacon_node/beacon_chain/src/data_availability_checker/error.rs b/beacon_node/beacon_chain/src/data_availability_checker/error.rs
index ab69a6298..2653c8486 100644
--- a/beacon_node/beacon_chain/src/data_availability_checker/error.rs
+++ b/beacon_node/beacon_chain/src/data_availability_checker/error.rs
@@ -1,7 +1,8 @@
 use kzg::{Error as KzgError, KzgCommitment};
+use strum::IntoStaticStr;
 use types::{BeaconStateError, ColumnIndex, Hash256};
 
-#[derive(Debug)]
+#[derive(Debug, IntoStaticStr)]
 pub enum Error {
     InvalidBlobs(KzgError),
     MissingBid(Hash256),
