diff --git a/beacon_node/beacon_chain/src/fetch_blobs/fetch_blobs_beacon_adapter.rs b/beacon_node/beacon_chain/src/fetch_blobs/fetch_blobs_beacon_adapter.rs
index f5ba647fc..b75fcdac5 100644
--- a/beacon_node/beacon_chain/src/fetch_blobs/fetch_blobs_beacon_adapter.rs
+++ b/beacon_node/beacon_chain/src/fetch_blobs/fetch_blobs_beacon_adapter.rs
@@ -1,8 +1,9 @@
-use crate::fetch_blobs::{EngineGetBlobsOutput, FetchEngineBlobError};
+use crate::data_column_verification::KzgVerifiedCustodyDataColumn;
+use crate::fetch_blobs::FetchEngineBlobError;
 use crate::observed_data_sidecars::ObservationKey;
 use crate::partial_data_column_assembler::PartialDataColumnAssembler;
 use crate::{AvailabilityProcessingStatus, BeaconChain, BeaconChainTypes};
-use execution_layer::json_structures::{BlobAndProofV1, BlobAndProofV2, BlobAndProofV3};
+use execution_layer::json_structures::{BlobAndProofV2, BlobAndProofV3};
 use kzg::Kzg;
 #[cfg(test)]
 use mockall::automock;
@@ -43,22 +44,6 @@ impl<T: BeaconChainTypes> FetchBlobsBeaconAdapter<T> {
             .cloned()
     }
 
-    pub(crate) async fn get_blobs_v1(
-        &self,
-        versioned_hashes: Vec<Hash256>,
-    ) -> Result<Vec<Option<BlobAndProofV1<T::EthSpec>>>, FetchEngineBlobError> {
-        let execution_layer = self
-            .chain
-            .execution_layer
-            .as_ref()
-            .ok_or(FetchEngineBlobError::ExecutionLayerMissing)?;
-
-        execution_layer
-            .get_blobs_v1(versioned_hashes)
-            .await
-            .map_err(FetchEngineBlobError::RequestFailed)
-    }
-
     pub(crate) async fn get_blobs_v2(
         &self,
         versioned_hashes: Vec<Hash256>,
@@ -91,17 +76,6 @@ impl<T: BeaconChainTypes> FetchBlobsBeaconAdapter<T> {
             .map_err(FetchEngineBlobError::RequestFailed)
     }
 
-    pub(crate) fn blobs_known_for_observation_key(
-        &self,
-        observation_key: ObservationKey,
-    ) -> Option<HashSet<u64>> {
-        self.chain
-            .observed_blob_sidecars
-            .read()
-            .known_for_observation_key(&observation_key)
-            .cloned()
-    }
-
     pub(crate) fn data_column_known_for_observation_key(
         &self,
         observation_key: ObservationKey,
@@ -113,12 +87,6 @@ impl<T: BeaconChainTypes> FetchBlobsBeaconAdapter<T> {
             .cloned()
     }
 
-    pub(crate) fn cached_blob_indexes(&self, block_root: &Hash256) -> Option<Vec<u64>> {
-        self.chain
-            .data_availability_checker
-            .cached_blob_indexes(block_root)
-    }
-
     pub(crate) fn cached_data_column_indexes(
         &self,
         block_root: &Hash256,
@@ -131,7 +99,7 @@ impl<T: BeaconChainTypes> FetchBlobsBeaconAdapter<T> {
         &self,
         slot: Slot,
         block_root: Hash256,
-        blobs: EngineGetBlobsOutput<T>,
+        blobs: Vec<KzgVerifiedCustodyDataColumn<T::EthSpec>>,
     ) -> Result<AvailabilityProcessingStatus, FetchEngineBlobError> {
         self.chain
             .process_engine_blobs(slot, block_root, blobs)
