diff --git a/beacon_node/execution_layer/Cargo.toml b/beacon_node/execution_layer/Cargo.toml
index a23ea948e..e37a85f14 100644
--- a/beacon_node/execution_layer/Cargo.toml
+++ b/beacon_node/execution_layer/Cargo.toml
@@ -10,14 +10,18 @@ alloy-primitives = { workspace = true }
 alloy-rlp = { workspace = true }
 alloy-rpc-types-eth = { workspace = true }
 arc-swap = "1.6.0"
+async-stream = "0.3"
+async-trait = "0.1"
 bls = { workspace = true }
 builder_client = { path = "../builder_client" }
 bytes = { workspace = true }
 eth2 = { workspace = true, features = ["events", "lighthouse", "network"] }
 ethereum_serde_utils = { workspace = true }
 ethereum_ssz = { workspace = true }
+ethereum_ssz_derive = { workspace = true }
 fixed_bytes = { workspace = true }
 fork_choice = { workspace = true }
+futures = { workspace = true }
 hash-db = "0.15.2"
 hash256-std-hasher = "0.15.2"
 hex = { workspace = true }
@@ -32,6 +36,7 @@ parking_lot = { workspace = true }
 pretty_reqwest_error = { workspace = true }
 rand = { workspace = true }
 reqwest = { workspace = true }
+reqwest-eventsource = "0.6"
 sensitive_url = { workspace = true }
 serde = { workspace = true }
 serde_json = { workspace = true }
