diff --git a/beacon_node/lighthouse_network/src/peer_manager/mod.rs b/beacon_node/lighthouse_network/src/peer_manager/mod.rs
index 6b5144fa6..161257979 100644
--- a/beacon_node/lighthouse_network/src/peer_manager/mod.rs
+++ b/beacon_node/lighthouse_network/src/peer_manager/mod.rs
@@ -593,6 +593,9 @@ impl<E: EthSpec> PeerManager<E> {
                     Protocol::BlobsByRange => PeerAction::MidToleranceError,
                     Protocol::PayloadEnvelopesByRange => PeerAction::MidToleranceError,
                     Protocol::PayloadEnvelopesByRoot => PeerAction::MidToleranceError,
+                    Protocol::ExecutionProofsByRange => PeerAction::MidToleranceError,
+                    Protocol::ExecutionProofsByRoot => PeerAction::MidToleranceError,
+                    Protocol::ExecutionProofStatus => PeerAction::MidToleranceError,
                     // Lighthouse does not currently make light client requests; therefore, this
                     // is an unexpected scenario. We do not ban the peer for rate limiting.
                     Protocol::LightClientBootstrap => return,
@@ -621,6 +624,9 @@ impl<E: EthSpec> PeerManager<E> {
                     Protocol::BlocksByHead => return,
                     Protocol::PayloadEnvelopesByRange => return,
                     Protocol::PayloadEnvelopesByRoot => return,
+                    Protocol::ExecutionProofsByRange => return,
+                    Protocol::ExecutionProofsByRoot => return,
+                    Protocol::ExecutionProofStatus => return,
                     Protocol::BlobsByRange => return,
                     Protocol::BlobsByRoot => return,
                     Protocol::DataColumnsByRoot => return,
@@ -647,6 +653,9 @@ impl<E: EthSpec> PeerManager<E> {
                     Protocol::BlocksByHead => PeerAction::MidToleranceError,
                     Protocol::PayloadEnvelopesByRange => PeerAction::MidToleranceError,
                     Protocol::PayloadEnvelopesByRoot => PeerAction::MidToleranceError,
+                    Protocol::ExecutionProofsByRange => PeerAction::MidToleranceError,
+                    Protocol::ExecutionProofsByRoot => PeerAction::MidToleranceError,
+                    Protocol::ExecutionProofStatus => PeerAction::MidToleranceError,
                     Protocol::BlobsByRange => PeerAction::MidToleranceError,
                     Protocol::BlobsByRoot => PeerAction::MidToleranceError,
                     Protocol::DataColumnsByRoot => PeerAction::MidToleranceError,
