diff --git a/beacon_node/lighthouse_network/src/types/globals.rs b/beacon_node/lighthouse_network/src/types/globals.rs
index df8dbdc55..0a9642bd8 100644
--- a/beacon_node/lighthouse_network/src/types/globals.rs
+++ b/beacon_node/lighthouse_network/src/types/globals.rs
@@ -226,6 +226,7 @@ impl<E: EthSpec> NetworkGlobals<E> {
     pub fn as_topic_config(&self) -> TopicConfig {
         TopicConfig {
             enable_light_client_server: self.config.enable_light_client_server,
+            enable_execution_proof: self.config.enable_execution_proof,
             subscribe_all_subnets: self.config.subscribe_all_subnets,
             sampling_subnets: self.sampling_subnets.read().clone(),
         }
