diff --git a/beacon_node/network/src/metrics.rs b/beacon_node/network/src/metrics.rs
index 4b34d7bfc..c043133ce 100644
--- a/beacon_node/network/src/metrics.rs
+++ b/beacon_node/network/src/metrics.rs
@@ -128,13 +128,6 @@ pub static BEACON_PROCESSOR_GOSSIP_BLOCK_EARLY_SECONDS: LazyLock<Result<Histogra
         )
     },
 );
-pub static BEACON_PROCESSOR_GOSSIP_BLOB_VERIFIED_TOTAL: LazyLock<Result<IntCounter>> =
-    LazyLock::new(|| {
-        try_create_int_counter(
-            "beacon_processor_gossip_blob_verified_total",
-            "Total number of gossip blob verified for propagation.",
-        )
-    });
 pub static BEACON_PROCESSOR_GOSSIP_DATA_COLUMN_SIDECAR_VERIFIED_TOTAL: LazyLock<
     Result<IntCounter>,
 > = LazyLock::new(|| {
@@ -600,12 +593,6 @@ pub static BEACON_BLOCK_DELAY_GOSSIP_ARRIVED_LATE_TOTAL: LazyLock<Result<IntCoun
 /*
  * Blob Delay Metrics
  */
-pub static BEACON_BLOB_DELAY_GOSSIP: LazyLock<Result<IntGauge>> = LazyLock::new(|| {
-    try_create_int_gauge(
-        "beacon_blob_delay_gossip_last_delay",
-        "The first time we see this blob as a delay from the start of the slot",
-    )
-});
 
 pub static BEACON_DATA_COLUMN_GOSSIP_PROPAGATION_VERIFICATION_DELAY_TIME: LazyLock<
     Result<Histogram>,
@@ -664,14 +651,6 @@ pub static BEACON_USEFUL_FULL_COLUMNS_RECEIVED_TOTAL: LazyLock<Result<IntCounter
         )
     });
 
-pub static BEACON_BLOB_DELAY_GOSSIP_VERIFICATION: LazyLock<Result<IntGauge>> = LazyLock::new(
-    || {
-        try_create_int_gauge(
-            "beacon_blob_delay_gossip_verification",
-            "Keeps track of the time delay from the start of the slot to the point we propagate the blob",
-        )
-    },
-);
 pub static BEACON_BLOB_DELAY_FULL_VERIFICATION: LazyLock<Result<IntGauge>> = LazyLock::new(|| {
     try_create_int_gauge(
         "beacon_blob_last_full_verification_delay",
@@ -695,15 +674,6 @@ pub static BEACON_BLOB_RPC_SLOT_START_DELAY_TIME: LazyLock<Result<Histogram>> =
     },
 );
 
-pub static BEACON_BLOB_GOSSIP_ARRIVED_LATE_TOTAL: LazyLock<Result<IntCounter>> = LazyLock::new(
-    || {
-        try_create_int_counter(
-            "beacon_blob_gossip_arrived_late_total",
-            "Count of times when a gossip blob arrived from the network later than the attestation deadline.",
-        )
-    },
-);
-
 /*
  * Light client update reprocessing queue metrics.
  */
