diff --git a/beacon_node/network/src/sync/backfill_sync/mod.rs b/beacon_node/network/src/sync/backfill_sync/mod.rs
index 0f80138d2..16d5011ad 100644
--- a/beacon_node/network/src/sync/backfill_sync/mod.rs
+++ b/beacon_node/network/src/sync/backfill_sync/mod.rs
@@ -81,6 +81,7 @@ impl<E: EthSpec> BatchConfig for BackFillBatchConfig<E> {
 }
 
 /// Return type when attempting to start the backfill sync process.
+#[cfg_attr(feature = "disable-backfill", allow(dead_code))]
 pub enum SyncStart {
     /// The chain started syncing or is already syncing.
     Syncing {
@@ -159,6 +160,7 @@ pub struct BackFillSync<T: BeaconChainTypes> {
     network_globals: Arc<NetworkGlobals<T::EthSpec>>,
 }
 
+#[cfg_attr(feature = "disable-backfill", allow(dead_code))]
 impl<T: BeaconChainTypes> BackFillSync<T> {
     pub fn new(
         beacon_chain: Arc<BeaconChain<T>>,
@@ -1210,6 +1212,7 @@ impl<T: BeaconChainTypes> BackFillSync<T> {
 }
 
 /// Error kind for attempting to restart the sync from beacon chain parameters.
+#[cfg_attr(feature = "disable-backfill", allow(dead_code))]
 enum ResetEpochError {
     /// The chain has already completed.
     SyncCompleted,
@@ -1247,7 +1250,7 @@ mod tests {
             let peer_id = network_globals
                 .peers
                 .write()
-                .__add_connected_peer_testing_only(
+                .__add_connected_peer_with_custody_subnets(
                     true,
                     &beacon_chain.spec,
                     k256::ecdsa::SigningKey::random(&mut rng).into(),
