diff --git a/consensus/types/src/data/partial_data_column_sidecar.rs b/consensus/types/src/data/partial_data_column_sidecar.rs
index c0e713b4b..e70901d76 100644
--- a/consensus/types/src/data/partial_data_column_sidecar.rs
+++ b/consensus/types/src/data/partial_data_column_sidecar.rs
@@ -69,7 +69,7 @@ impl<E: EthSpec> PartialDataColumnSidecar<E> {
             .count();
         self.column
             .get(storage_idx)
-            .and_then(|cell| self.kzg_proofs.get(storage_idx).map(|proof| (cell, proof)))
+            .zip(self.kzg_proofs.get(storage_idx))
     }
 
     /// Creates a reference to this sidecar containing only the blob indices for which the passed
