diff --git a/testing/ef_tests/src/cases.rs b/testing/ef_tests/src/cases.rs
index b2e027635..b2386f6fa 100644
--- a/testing/ef_tests/src/cases.rs
+++ b/testing/ef_tests/src/cases.rs
@@ -20,6 +20,7 @@ mod fork_choice;
 mod genesis_initialization;
 mod genesis_validity;
 mod get_custody_groups;
+mod gossip_validation;
 mod kzg_blob_to_kzg_commitment;
 mod kzg_compute_blob_kzg_proof;
 mod kzg_compute_cells;
@@ -57,6 +58,7 @@ pub use fork::ForkTest;
 pub use genesis_initialization::*;
 pub use genesis_validity::*;
 pub use get_custody_groups::*;
+pub use gossip_validation::*;
 pub use kzg_blob_to_kzg_commitment::*;
 pub use kzg_compute_blob_kzg_proof::*;
 pub use kzg_compute_cells::*;
