diff --git a/testing/simulator/src/basic_sim.rs b/testing/simulator/src/basic_sim.rs
index 79581ee52..aa7896e20 100644
--- a/testing/simulator/src/basic_sim.rs
+++ b/testing/simulator/src/basic_sim.rs
@@ -1,4 +1,5 @@
 use crate::local_network::LocalNetworkParams;
+use crate::local_network::NodeType;
 use crate::local_network::TERMINAL_BLOCK;
 use crate::{LocalNetwork, checks};
 use clap::ArgMatches;
@@ -30,9 +31,10 @@ const ALTAIR_FORK_EPOCH: u64 = 0;
 const BELLATRIX_FORK_EPOCH: u64 = 0;
 const CAPELLA_FORK_EPOCH: u64 = 0;
 const DENEB_FORK_EPOCH: u64 = 0;
-const ELECTRA_FORK_EPOCH: u64 = 2;
-// const FULU_FORK_EPOCH: u64 = 3;
-// const GLOAS_FORK_EPOCH: u64 = 4;
+const ELECTRA_FORK_EPOCH: u64 = 0;
+const FULU_FORK_EPOCH: u64 = 0;
+// TODO(gloas): enable Gloas in simulator, current blocker is lack of data column gossip verification
+// const GLOAS_FORK_EPOCH: u64 = 2;
 
 const SUGGESTED_FEE_RECIPIENT: [u8; 20] =
     [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1];
@@ -171,8 +173,8 @@ pub fn run_basic_sim(matches: &ArgMatches) -> Result<(), String> {
     let genesis_delay = GENESIS_DELAY;
 
     // Convenience variables. Update these values when adding a newer fork.
-    let latest_fork_version = spec.electra_fork_version;
-    let latest_fork_start_epoch = ELECTRA_FORK_EPOCH;
+    let latest_fork_version = spec.fulu_fork_version;
+    let latest_fork_start_epoch = FULU_FORK_EPOCH;
 
     let mut slot_duration_ms = spec.get_slot_duration().as_millis() as u64;
     slot_duration_ms /= speed_up_factor;
@@ -187,6 +189,7 @@ pub fn run_basic_sim(matches: &ArgMatches) -> Result<(), String> {
     spec.capella_fork_epoch = Some(Epoch::new(CAPELLA_FORK_EPOCH));
     spec.deneb_fork_epoch = Some(Epoch::new(DENEB_FORK_EPOCH));
     spec.electra_fork_epoch = Some(Epoch::new(ELECTRA_FORK_EPOCH));
+    spec.fulu_fork_epoch = Some(Epoch::new(FULU_FORK_EPOCH));
     let spec = Arc::new(spec);
     env.eth2_config.spec = spec.clone();
 
@@ -208,8 +211,11 @@ pub fn run_basic_sim(matches: &ArgMatches) -> Result<(), String> {
                 LocalNetworkParams {
                     validator_count: total_validator_count,
                     node_count,
-                    extra_nodes,
+                    extra_nodes: 0,
                     proposer_nodes,
+                    proof_generator_nodes: 0,
+                    proof_verifier_nodes: 0,
+                    delayed_nodes: extra_nodes,
                     genesis_delay,
                 },
                 context.clone(),
@@ -220,7 +226,11 @@ pub fn run_basic_sim(matches: &ArgMatches) -> Result<(), String> {
         // Add nodes to the network.
         for _ in 0..node_count {
             network
-                .add_beacon_node(beacon_config.clone(), mock_execution_config.clone(), false)
+                .add_beacon_node(
+                    beacon_config.clone(),
+                    mock_execution_config.clone(),
+                    NodeType::Default,
+                )
                 .await?;
         }
 
@@ -230,7 +240,11 @@ pub fn run_basic_sim(matches: &ArgMatches) -> Result<(), String> {
         for _ in 0..proposer_nodes {
             println!("Adding a proposer node");
             network
-                .add_beacon_node(beacon_config.clone(), mock_execution_config.clone(), true)
+                .add_beacon_node(
+                    beacon_config.clone(),
+                    mock_execution_config.clone(),
+                    NodeType::Proposer,
+                )
                 .await?;
         }
 
@@ -261,7 +275,7 @@ pub fn run_basic_sim(matches: &ArgMatches) -> Result<(), String> {
                             .await
                     } else {
                         network_1
-                            .add_validator_client(validator_config, i, files)
+                            .add_validator_client(validator_config, i, files, NodeType::Default)
                             .await
                     }
                     .expect("should add validator");
