diff --git a/testing/simulator/src/fallback_sim.rs b/testing/simulator/src/fallback_sim.rs
index 06f4478c5..3d872c1a2 100644
--- a/testing/simulator/src/fallback_sim.rs
+++ b/testing/simulator/src/fallback_sim.rs
@@ -1,4 +1,4 @@
-use crate::local_network::LocalNetworkParams;
+use crate::local_network::{LocalNetworkParams, NodeType};
 use crate::{LocalNetwork, checks};
 use clap::ArgMatches;
 
@@ -25,11 +25,12 @@ const END_EPOCH: u64 = 16;
 const GENESIS_DELAY: u64 = 38;
 const ALTAIR_FORK_EPOCH: u64 = 0;
 const BELLATRIX_FORK_EPOCH: u64 = 0;
-const CAPELLA_FORK_EPOCH: u64 = 1;
-const DENEB_FORK_EPOCH: u64 = 2;
-// const ELECTRA_FORK_EPOCH: u64 = 3;
-// const FULU_FORK_EPOCH: u64 = 4;
-// const GLOAS_FORK_EPOCH: u64 = 5;
+const CAPELLA_FORK_EPOCH: u64 = 0;
+const DENEB_FORK_EPOCH: u64 = 0;
+const ELECTRA_FORK_EPOCH: u64 = 0;
+const FULU_FORK_EPOCH: u64 = 0;
+// TODO(gloas): enable Gloas in simulator, current blocker is lack of data column gossip verification
+// const GLOAS_FORK_EPOCH: u64 = 2;
 
 // Since simulator tests are non-deterministic and there is a non-zero chance of missed
 // attestations, define an acceptable network-wide attestation performance.
@@ -191,8 +192,8 @@ pub fn run_fallback_sim(matches: &ArgMatches) -> Result<(), String> {
     spec.bellatrix_fork_epoch = Some(Epoch::new(BELLATRIX_FORK_EPOCH));
     spec.capella_fork_epoch = Some(Epoch::new(CAPELLA_FORK_EPOCH));
     spec.deneb_fork_epoch = Some(Epoch::new(DENEB_FORK_EPOCH));
-    //spec.electra_fork_epoch = Some(Epoch::new(ELECTRA_FORK_EPOCH));
-    //spec.fulu_fork_epoch = Some(Epoch::new(FULU_FORK_EPOCH));
+    spec.electra_fork_epoch = Some(Epoch::new(ELECTRA_FORK_EPOCH));
+    spec.fulu_fork_epoch = Some(Epoch::new(FULU_FORK_EPOCH));
     let spec = Arc::new(spec);
     env.eth2_config.spec = spec.clone();
 
@@ -218,6 +219,9 @@ pub fn run_fallback_sim(matches: &ArgMatches) -> Result<(), String> {
                     node_count,
                     extra_nodes: 0,
                     proposer_nodes: 0,
+                    proof_generator_nodes: 0,
+                    proof_verifier_nodes: 0,
+                    delayed_nodes: 0,
                     genesis_delay,
                 },
                 context.clone(),
@@ -228,7 +232,11 @@ pub fn run_fallback_sim(matches: &ArgMatches) -> Result<(), String> {
         // Add nodes to the network.
         for _ in 0..node_count {
             network
-                .add_beacon_node(beacon_config.clone(), mock_execution_config.clone(), false)
+                .add_beacon_node(
+                    beacon_config.clone(),
+                    mock_execution_config.clone(),
+                    NodeType::Default,
+                )
                 .await?;
         }
 
