diff --git a/validator_client/http_api/src/tests.rs b/validator_client/http_api/src/tests.rs
index 5cb631983..404fc6c0f 100644
--- a/validator_client/http_api/src/tests.rs
+++ b/validator_client/http_api/src/tests.rs
@@ -132,6 +132,7 @@ impl ApiTester {
             },
             sse_logging_components: None,
             slot_clock: slot_clock.clone(),
+            proof_service: None,
         });
         let ctx = context.clone();
         let (listening_socket, server) =
