diff --git a/validator_client/validator_metrics/src/lib.rs b/validator_client/validator_metrics/src/lib.rs
index 46a86381f..0d1fb4a8c 100644
--- a/validator_client/validator_metrics/src/lib.rs
+++ b/validator_client/validator_metrics/src/lib.rs
@@ -126,6 +126,13 @@ pub static SIGNED_VALIDATOR_REGISTRATIONS_TOTAL: LazyLock<Result<IntCounterVec>>
             &["status"],
         )
     });
+pub static SIGNED_EXECUTION_PROOFS_TOTAL: LazyLock<Result<IntCounterVec>> = LazyLock::new(|| {
+    try_create_int_counter_vec(
+        "vc_signed_execution_proofs_total",
+        "Total count of ExecutionProof signings",
+        &["status"],
+    )
+});
 pub static DUTIES_SERVICE_TIMES: LazyLock<Result<HistogramVec>> = LazyLock::new(|| {
     try_create_histogram_vec(
         "vc_duties_service_task_times_seconds",
